daffodil-io/src/main/scala/org/apache/daffodil/processors/charset/BitsCharsetDecoder.scala

```diff
@@ -32,6 +31,13 @@ import org.apache.daffodil.util.MaybeChar
 class BitsCharsetDecoderMalformedException(val malformedBits: Int)
   extends ThinException
 
+class BitsCharsetDecoderUnalignedCharDecodeException(val bitAlignment1b: Long)
+  extends ThinException {
+  override def getMessage(): String = {
+    s"Charset not byte aligned. Bit alignment (1b): ${bitAlignment1b}."
```

Comment:
Is reporting the desired `bitAlignment1b` more useful for debugging than reporting the current value of the stream's `bitPos1b`?  Maybe the caller on line 141 should pass `dis.bitPos1b` instead of `dis.bitPos1b % 8` and this exception should capture the value as `bitPos1b` instead of `bitAlignment1b`?

Reply:
Arguably we should save the bitPos1b and have the message refer to both the bitAlignment1b, the bitPos1b and the bytePos1b since it depends on the data what is useful to the user. 
I will improve this.